File: tests/conformance/fixtures/typescript/18-edge-provides-empty.ts
import { Pipeline } from '../../../../sdk/typescript/src/index';

const p = new Pipeline();

p.task('build').run('make build').provides('artifact', '');

p.task('migrate').run('dbmate up').provides('db-ready');

p.task('package').run('docker build').provides('image', 'myapp:latest').after('build');

p.emit();

File: tests/conformance/fixtures/rust/18-edge-provides-empty.rs
use sykli::Pipeline;

fn main() {
    let mut p = Pipeline::new();
    p.task("build").run("make build").provides("artifact", None);
    p.task("migrate").run("dbmate up").provides("db-ready", None);
    p.task("package").run("docker build").provides("image", Some("myapp:latest")).after(&["build"]);
    p.emit();
}
